spec/system/grant_submissions/grant_submission_responses_spec.rb

```diff
@@ -8,137 +8,342 @@ def successful_application_message
+            context 'submitting' do
+              context 'requires response mandatory short_text question' do
+                scenario 'no answer for required short text' do
+                  click_link 'Edit'
+                  expect(page).to have_current_path edit_grant_submission_path(@grant, @submission)
+                  find_field('Number Question').set(Faker::Number.number(digits: 10))
+                  find_field('Long Text Question').set(Faker::Lorem.paragraph_by_chars(number: 1000))
+                  click_button 'Submit'
+                  expect(page).to have_content 'Please review the following error'
+                  expect(page).to have_content 'is invalid'
+                end
+
+                scenario 'with answer for required short text' do
+                  click_link 'Edit'
+                  expect(page).to have_current_path edit_grant_submission_path(@grant, @submission)
+                  find_field('Number Question').set(Faker::Number.number(digits: 10))
+                  find_field('Long Text Question').set(Faker::Lorem.paragraph_by_chars(number: 1000))
+                  find_field('Short Text Question', with:'').set(Faker::Lorem.paragraph_by_chars(number: 200))
+                  click_button 'Submit'
+                  expect(page).to have_content 'You successfully applied.'
+                  expect(page).to have_current_path profile_submissions_path
+                  expect(page).to have_content GrantSubmission::Submission.last.title
+                  expect(page).not_to have_content 'Edit'
+                end
+              end
+            end
+          end
+        end
+      end
+    end
+
+    describe 'as a grant admin' do
+      before(:each) do
+        @applicant  = create(:user)
+        @admin      = @grant.administrators.first
+
+        @submission = create(:draft_submission_with_responses,
+                                  grant:      @grant,
+                                  form:       @grant.form,
+                                  created_id: @applicant.id,
+                                  title:      Faker::Lorem.sentence,
+                                  state:      'draft')
+              short_text_question = @grant.questions.where(response_type: 'short_text').first
+              short_text_question.update_attribute(:is_mandatory, true)
+
+              short_text_response = @submission.responses.where(question: short_text_question).first
+              short_text_response.update_attribute(:string_val, '')
+
+        login_as(@admin)
+        visit grant_path(@grant)
+      end
+
+      context 'states' do
+        let(:file_upload_question) {@grant.questions.find_by(response_type: 'file_upload')}
+
+        context 'single required question' do
+
+          context 'on update' do
+            context 'saving as draft' do
+              context 'requires response mandatory short_text question' do
+                scenario 'no answer for required short text' do
```

Comment:
I think the context and scenario descriptions need to be clarified.

In 295 'requires response mandatory short_text question' doesn't read well. 
In 296 'no response' may be sufficient.